cypress/integration/Popup.spec.ts

```diff
@@ -117,6 +117,37 @@ describe('Popup', () => {
         it.skip('should close Popup 1', () => {
           cy.findAllByLabelText('Popup 1').should('not.exist');
         });
+
+        context('then Open Popup 3 button is clicked', () => {
+          const Popup3Title = 'Popup 3 (Not hidable on outside click)';
+          beforeEach(() => {
+            cy.findByText('Open Popup 3').click();
+          });
+
+          it('should open Popup 3 in front of Popup 2', () => {
+            cy.findByLabelText(Popup3Title).should('be.visible');
+            cy.findByLabelText('Popup 2').should('be.visible');
+
+            cy.findByLabelText(Popup3Title).should(h.popup.beOnTopOfLabelledByText('Popup 2'));
+          });
+
+          context('then the close button in Popup 3 is clicked', () => {
+            beforeEach(() => {
+              cy.findByLabelText(Popup3Title)
+                .parentsUntil('[data-popper-placement]')
+                .find('[data-close]')
+                .click();
```

Comment:
This selector group is a bit weird since too much non-semantic DOM info must be known. Cypress testing library gives us some good selectors for what we need. What about the following:
```suggestion
              cy.findByRole('dialog', {name: Popup3Title})
                .findByRole('button', {name: 'Close'})
                .click();
```

The other specs should be updated to use `findByRole` as well. `name` wasn't a scoping option at the time these specs were written, but I think it makes it more clear.